compiler/rustc_codegen_llvm/src/asm.rs

```diff
@@ -874,3 +903,35 @@ fn llvm_fixup_output_type(
         _ => layout.llvm_type(cx),
     }
 }
+
+// Valid llvm symbols: https://llvm.org/docs/AMDGPUOperandSyntax.html#symbols
+// First char is [a-zA-Z_.]
+fn llvm_ident_start(c: char) -> bool {
+    ('a'..='z').contains(&c) || ('A'..='Z').contains(&c) || '_' == c || '.' == c
+}
+
+// All subsequent chars are [a-zA-Z0-9_$.@]
+fn llvm_ident_continuation(c: char) -> bool {
+    ('a'..='z').contains(&c)
+        || ('A'..='Z').contains(&c)
+        || ('0'..='9').contains(&c)
+        || '_' == c
+        || '$' == c
+        || '.' == c
+        || '@' == c
```

Comment:
A `?` mark is a valid continuation character according to clang: https://rust.godbolt.org/z/EsP4sW, though it does appear like a bug and is only intended to be allowed when emulating the microsoft `cl`. Nevertheless it is something that LLVM currently will let through.